source/Politics.cpp

```diff
@@ -258,6 +258,36 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 
 		int failedMissions = 0;
 
+		// Illegal passengers can only be detected by planetary security.
+		if(!scan)
+		{
+			int64_t fine = ship->Cargo().IllegalPassengersFine(gov, player);
+			if((fine > maxFine && maxFine >= 0) || fine < 0)
+			{
+				maxFine = fine;
+				reason = " for carrying illegal passengers.";
+
+				for(const Mission &mission : player.Missions())
+				{
+					if(mission.IsFailed(player))
+						continue;
+
+					// IllegalCargoMessage applies to both cargo and passengers.
+					string illegalCargoMessage = mission.IllegalCargoMessage();
```

Comment:
I feel like we should just rename this function at this point. Thoughts?

Reply:
Would "FineMessage" be OK?

Reply:
Yeah.